File: back-end/src/main/java/es/ull/project/application/usecase/workflow/WorkFlowCreateUseCase.java
package es.ull.project.application.usecase.workflow;

import java.util.List;

import es.ull.project.domain.entity.WorkFlow;
import es.ull.project.domain.valueobject.WorkFlowName;
import es.ull.project.domain.valueobject.Node;

public interface WorkFlowCreateUseCase {

    /**
     * Create an workflow
     * 
     * @param sourceName
     * @param sourceType
     * @return workflow created
     */
    WorkFlow createWorkFlow(WorkFlowName name, List<Node> nodes, boolean enabled);
}

File: front-end/src/application/usecase/workflow/workflow-create-use-case.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError } from '@ull-tfg/ull-tfg-typescript';
import { WorkFlow } from '@/domain/entity/workflow';
import { WorkFlowName } from '@/domain/value-object/workflow-name';
import { Node } from '@/domain/value-object/node';

export interface WorkFlowCreateUseCase {
    
    /**
     * Create a workflow
     * 
     * @param workFlowName - Name of the workflow
     * @param nodes - Map of operations and their corresponding predicates
     * @param creationDate - Date of creation of the workflow
     * @returns The created workflow, if successful
     */
    createWorkFlow(
        workFlowName: WorkFlowName,
        nodes: Node[],
        creationDate: Date | string,
        enabled: boolean
    ): Promise<Either<DataError, WorkFlow>>;
}
